src/rust/engine/process_executor/src/main.rs

```diff
@@ -263,19 +296,27 @@ fn main() {
           args.value_of("cache-key-gen-version").map(str::to_owned),
           remote_instance_arg,
           oauth_bearer_token,
-          store,
+          store.clone(),
           timer_thread,
         )
         .expect("Could not initialize remote execution client"),
       ) as Box<dyn process_execution::CommandRunner>
     }
     None => Box::new(process_execution::local::CommandRunner::new(
-      store, pool, work_dir, true,
+      store.clone(),
+      pool,
+      work_dir,
+      true,
     )) as Box<dyn process_execution::CommandRunner>,
   };
   let mut rt = tokio::runtime::Runtime::new().unwrap();
   let result = rt.block_on(runner.run(request)).unwrap();
 
+  if let Some(materialize_directory) = args.value_of("materialize-output-to").map(PathBuf::from) {
```

Comment:
I'd probably call this variable `output` or `materialize_directory_destination`, rather than `materialize_directory`